Vagrantfile

```diff
@@ -0,0 +1,134 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Simulate ActiveModel::Type::Boolean.new.cast(value)
+$FALSE_VALUES = [false, 0, "0", "f", "F", "false", "FALSE", "off", "OFF"]
+
+# Should Vagrant modify /etc/exports automatically?
+$nfs_export = $FALSE_VALUES.none? ENV['VAGRANT_NFS_EXPORT']
+
+prov_env = {
+  "DEBIAN_FRONTEND" => "noninteractive",
+}
+
+$init = <<SCRIPT
+  set -e
+  apt -y update
+  apt -y install \
+    build-essential git tmux vim \
+    autoconf automake libtool \
+    libssl-dev net-tools
+SCRIPT
+
+$python = <<SCRIPT
+  set -e
+  apt -y install \
+    python3-all python3-dev \
+    python3-pip python3-venv
+SCRIPT
+
+$docker_ce = <<SCRIPT
+  set -e
+  apt -y install \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    software-properties-common
+  curl -fsSL https://download.docker.com/linux/ubuntu/gpg |
+    apt-key add -
+  add-apt-repository \
+    "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) \
+    stable"
+  apt -y install \
+    containerd.io \
+    docker-ce \
+    docker-ce-cli
+SCRIPT
+
+$docker_compose = <<SCRIPT
+  set -e
+  curl \
+    -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" \
```

Comment:
Instead of hard coding a version, better to download the latest, e.g.:

```
docker_compose_version=$(curl -sLf https://api.github.com/repos/docker/compose/releases/latest | jq -r '.name')
curl -L "https://github.com/docker/compose/releases/download/${docker_compose_version}/docker-compose-$(uname -s)-$(uname -m)"
```

Reply:
I tried something like that earlier, but my github-fu wasn't good enough - thanks for the soln